src/preprocess.py

```diff
@@ -36,6 +75,79 @@ def video_to_frames(input_video,
 
     video.output(out_file).run(overwrite_output=True)
 
+def crop_image(input_path,
+               output_path,
+               height,
+               width=None,
+               x_offset=0,
+               y_offset=0):
+
+
+    image = Image.open(input_path)
+    orig_width, orig_height = image.size
+    orig_aspect_ratio = orig_width / orig_height
+    scaled_width = int(height * orig_aspect_ratio)
+    
+
+    image = image.resize((scaled_width, height), Image.Resampling.BICUBIC)
+    
+
+    if width is not None:
+        image = image.crop((x_offset, y_offset, x_offset + width, y_offset + height))
+
+    image.save(output_path)
+    
+def crop_frames(input_dir, 
+                    output_dir, 
+                    height, 
+                    width=None, 
+                    x_offset=0, 
+                    y_offset=0,
+                    should_skip_existing=True):
+
+    Path(output_dir).mkdir(parents=True, exist_ok=True)
+
+    for path in Path(input_dir).iterdir():
+        if path.is_file() and path.suffix == ".png":
+            output_path = f"{output_dir}/{path.name}"
+
+            if should_skip_existing and Path(output_path).is_file():
+                continue
+            
+            crop_image(path, output_path, height, width, x_offset, y_offset)
```

Comment:
The crop_frames are nice have but hopefully the crop code in the extraction will suffice because it is much faster to do all the resizing/cropping when converting the frames from the video, first doing a second pass afterwards.